### DISCUSSION TITLE
Could pos1 & end key escape Code Complete? (or add editor option for it)

### DISCUSSION THREAD

**Discussion by single-hyphen** [godotengine/godot-proposals#6488] [Editor]
I often move through my code using arrow-keys, but also pos1 and end. 
When I type something, for example replacing a variable in the center of a line the code complete will catch the first end-key press, jumping to the end of the list instead of moving my cursor to the end of the line. Now I almost never use the code complete while it has a large number of options left. So to me it would be incredibly rare for that to be my intention. 
For pos1 this is not an issue since the code complete window starts at its first element. So if it opens and you press pos1 it sets the cursor anyways. 

So my personal preference would be that when the code complete window opens and you press end once, your cursor is set to the end of line, instead of keeping the cursor and just scrolling the code complete window. 

Not sure how many people care about this. I assume not that many, but I'd like to hear if someone cares or how high the chances are for this if I'd post it as a feature request. I know this might sound like a fringe case, but it happens a lot for me. 

Now from Godot 4 I personally also rather not have the code complete try to be too smart in some cases. Like when you want to create a variable and set it to zero, you are quickly recommended stuff like "KEY_0". To me this is honestly a step back from the Godot 3 where you can just type an integer without then having to close out of the code complete window.

https://user-images.githubusercontent.com/15028607/224572815-b9531c88-5c68-4940-905e-f0d1e54a8680.mp4

After I replace "old" with "new" I pressed the end-key to move back to the end of the line. Instead I just jump to the end of code complete.

**Comment by bend-n** [comment]
the `KEY_0` issue has been fixed btw https://github.com/godotengine/godot/pull/74466.
And what is a pos1 key?

**Reply by zinnschlag** [reply]
Name of the home key on some non-English keyboards.
